Source/Scene/ModelExperimental/ModelExperimentalStatistics.js

```diff
@@ -49,20 +50,51 @@ export default function ModelExperimentalStatistics() {
   this.texturesByteLength = 0;
 
   /**
-   * Total size of property tables, including the batch textures used for
+   * Total size of property tables. This excludes the batch textures used for
    * picking and styling.
    *
    * @type {Number}
    * @private
    */
   this.propertyTablesByteLength = 0;
 
-  // Sets of buffers and textures we've already counted, so we don't
-  // double-count cached assets.
-  this.bufferIdSet = {};
-  this.textureIdSet = {};
+  // Sets of buffers and textures that have already been counted.
+  // This is to prevent double-counting cached assets.
+  this._bufferIdSet = {};
+  this._textureIdSet = {};
+
+  // Associated array of batch textures that have already been counted.
+  // This allows for quick look-up to check if a texture has been counted,
+  // while also allowing for dynamic texture counting.
+  this._batchTextureIdMap = new AssociativeArray();
 }
 
+Object.defineProperties(ModelExperimentalStatistics.prototype, {
+  /**
+   * Total size of the batch textures used for picking and styling.
```

Comment:
I would add a sentence saying that this loops over batch textures to ensure the value is up-to-date (as batch textures are asynchronous)

Reply:
Ah I see you did mention it in `addBatchTexture`, but wouldn't hurt to mention it here briefly